modules/product/src/ProductAttributeFieldManager.php

```diff
@@ -143,7 +143,7 @@ protected function buildFieldMap() {
         $handler_settings = $definition->getSetting('handler_settings');
         $component = $form_display->getComponent($field_name);
 
-        $field_map[$bundle][] = [
+        $field_map[$bundle][$field_name] = [
```

Comment:
See https://github.com/drupalcommerce/commerce/pull/368/files#diff-354d71e20ba2cfced9d85f311aefa77cR303

So we can foreach, use key for field name and still have attribute name for that field (so we can load it, see how it should render.) I feel there is a better approach, but last night it was +1 to get it done.